### ISSUE TITLE
points conveniance function does not evaluate lazily

### ISSUE THREAD

**Issue by BENR0** [pygeos/pygeos#397]
If point geometries are created with the the `points` conveniance function and dask arrays the geometries are not returned lazily because in https://github.com/pygeos/pygeos/blob/ea76ac9b287f85bf120eeb3d326d1251eff1f995/pygeos/creation.py#L67
`np.broadcast` is called on the arrays.

I worked around this by directly using `pygeos.lib.points` which works fine unless a `dask.distributed` cluster is used in which case a critical error `Failed to deserialize` is raised by `distributed`.

As a sidenote: using dask arrays and distributed to create box polygons works fine.

I use pygeos 0.10.2

**Comment by caspervdw** [comment]
Thanks for your report @BENR0 and great to hear that people are trying to combine PyGEOS and dask.

If I read this piece of code correctly, you could workaround by providing the points as a single coordinate array:

```
>>> pygeos.points([[0, 0], [1, 1]])
```

As opposed to providing x and y separately.

I do think we should do something about it. I am tending to break the API and remove the separate `y, z=None` in the call signatures (of course first with deprecation warnings). @jorisvandenbossche  Do you foresee any downstream issues?

**Comment by caspervdw** [comment]
> unless a dask.distributed cluster is used in which case a critical error Failed to deserialize is raised by distributed.

Could you elaborate on the "critical error"? `pygeos.Geometry` should deserialize fine.

**Comment by martinfleis** [comment]
> I am tending to break the API and remove the separate y, z=None in the call signatures (of course first with deprecation warnings). @jorisvandenbossche Do you foresee any downstream issues?

We would need to change the implementation of `points_from_xy` we have in geopandas. And I assume that we're not the only package using `points` in this way, so I assume it could break some stuff. I would therefore go for a longer deprecation period (across at least two versions?).

https://github.com/geopandas/geopandas/blob/64c5813e8b91231dfeed5268b82c2fced5cc1844/geopandas/_vectorized.py#L241-L255

If you want to do that deprecation, it may be wise to switch to a single array in geopandas before we'll release 0.10 this week.

**Comment by caspervdw** [comment]
So in that case I think it is not worth the fuss. Dask users should use the “coords” argument only. Let’s see if we can document this somewhere.

**Comment by BENR0** [comment]
Thanks for all the replies.

> As opposed to providing x and y separately

Yes I have tried that and weirdly could not get it to work before. But after trying again it works (probably one of those cases where the problem sits in front of the screen :-D). 

@caspervdw regarding the serialization issue I get:

```Python
AttributeError: Can't get attribute 'points' on <module '__main__' (built-in)>
distributed.core - ERROR - Can't get attribute 'points' on <module '__main__' (built-in)>
```
only if I use `dask.distributed.Client()`. With `processes=False` it works but I guess because there is no need for serialization in that case?!
Maybe this is also caused by myself. I will try this with a clean environment later and see if I can reproduce this.

**Comment by BENR0** [comment]
Ok I tried and still get the error. Just to make clear what I am actually doing :

```Python
import pygeos

ddf["geometry"] = pygeos.points(ddf[["lon", "lat"]].values)
```
where `ddf` is a dask dataframe.
Then I want to convert this to a dask-geodataframe and set the geometry column. The reason is that I tried to use it as a workaround due to a bug in `set_geometry()` in dask-geopandas which has already been fixed. So I do not rely on this anymore.
